sdk/samples/iot/hub/src/paho_iot_hub_c2d_example.c

```diff
@@ -290,39 +321,63 @@ static int subscribe()
   return MQTTCLIENT_SUCCESS;
 }
 
-static int on_received(void* context, char* topicName, int topicLen, MQTTClient_message* message)
+static void receive_messages()
 {
-  (void)context;
   int rc;
-
-  printf("Received a message from service.\n");
-  printf("Topic: %s\n", topicName);
-
-  if (topicLen == 0)
-  {
-    // The length of the topic if there are one or more NULL characters embedded in topicName,
-    // otherwise topicLen is 0.
-    topicLen = (int)strlen(topicName);
-  }
-  az_span topic_span = az_span_init((uint8_t*)topicName, topicLen);
-
-  // Parse the incoming message topic and check to make sure it is a C2D message
+  char* topic = NULL;
+  int topic_len = 0;
+  MQTTClient_message* message = NULL;
   az_iot_hub_client_c2d_request c2d_request;
-  if (az_failed(rc = az_iot_hub_client_c2d_parse_received_topic(&client, topic_span, &c2d_request)))
-  {
-    printf("Topic is not a C2D message, az_result return code %04x\n", rc);
-  }
-  else
+
+  // Wait until max # messages received or timeout to receive a single message expires.
+  for(uint8_t message_count = 0; message_count < MAX_MESSAGE_COUNT; message_count++)
   {
-    if (message->payloadlen)
+    if (((rc
+          = MQTTClient_receive(mqtt_client, &topic, &topic_len, &message, TIMEOUT_MQTT_RECEIVE_MS))
+          != MQTTCLIENT_SUCCESS)
+          && (MQTTCLIENT_TOPICNAME_TRUNCATED != rc))
+    {
+      LOG_ERROR("Failed to receive message: MQTTClient return code %d.", rc);
+      exit(rc);
+    }
+    else if (NULL == message)
```

Comment:
```suggestion
    else if (message == NULL)
```

Reply:
good timing :)  I JUST saw the note about this in the C guidelines! https://azure.github.io/azure-sdk/clang_implementation.html#clang-format-clang-no-yoda

Reply:
Reading the constant on the left-hand-side felt odd to me a well when I was glancing over your PR, earlier. I am glad we have it in the guidelines :)

Reply:
Having the constant on the left is a style used by some teams, as it defends the programmer from inadvertently assigning the constant to the variable.
Example:
A programmer intends to write `if (a == 5)`, but accidentally writes `if (a = 5)`.  The compiler will not catch this because it is a valid assignment.  However, if the rvalue (constant) is on the left and the programmer mistypes the intended expression `if (5 == a)` to be `if (5 = a)`, the compiler will catch this because the rvalue has no stored location in memory, hence assignment is not possible.  If both sides of the equivalency are lvalues, this approach is not applicable.